src/substrate/geneticTestingEvent.ts

```diff
@@ -84,24 +96,37 @@ export default class GeneticTestingEventHandler implements OnModuleInit {
       console.log(error);
     }
   }
-
-  onDnaSampleGenotypedSequenced(event) {
-    console.log('DnaSampleGenotypedSequenced! TODO: handle event');
+  
+  onDnaSampleResultReady(event) {
+    console.log('DnaSampleResultReady');
   }
 
-  onDnaSampleReviewed(event) {
-    console.log('DnaSampleReviewed! TODO: handle event');
+  onDnaTestResultSubmitted(event) {
+    console.log('DnaTestResultSubmitted');
   }
 
-  onDnaSampleComputed(event) {
-    console.log('DnaSampleComputed TODO: handle event');
-  }
+  async onDataStaked(event) {
+    console.log('DataStaked');
+    const dataStaked = event.data[0].toJSON();
+    const dataOrder = await (
+      await this.api.query.orders.orders(dataStaked.order_id)
```

Comment:
You should separate queries to the blockchain inside a `polkadot-providers` folder, please refer to frontend: https://github.com/debionetwork/debio-customer-ui/tree/development/src/common/lib/polkadot-provider